### DISCUSSION TITLE
bind socket to specified interface name

### DISCUSSION THREAD

**Discussion by nonikon** [libuv/libuv#3229] [Q&A] [answered]
somtimes we need to bind a socket to specified interface in tcp client, usually:
```
int fd = socket(...)
// eg: bind 'fd' to interface 'eth0'
setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, ...)
connect(fd, ...)
write(fd, ...)
```
libuv has no API about this, so I tried:
```
uv_tcp_init(..., &io)
setsockopt(io.io_watcher.fd, SOL_SOCKET, SO_BINDTODEVICE, ...)
uv_tcp_connect(..., &io, ...)
uv_write(...)
```
but, as 'uv_tcp_init' will not create socket, I got an EBADF errno in 'setsockopt'.
Is there any solution about this?
* **Version**: 1.41.0
* **Platform**: ubuntu 20.04

**Comment by santigimeno** [comment, marked as answer]
I think using [uv_tcp_open()](http://docs.libuv.org/en/v1.x/tcp.html#c.uv_tcp_open) is what you're looking for:
- Create a regular tcp socket with `socket()`.
- You can set the specific option to the socket
- `uv_tcp_init()` a new uv_tcp_handle
- `uv_tcp_open()` to associate the socket to the handle.
- call `uv_tcp_connect()`

**Reply by nonikon** [reply]
got it, thanks.
